src/releaseManager.ts

```diff
@@ -1,58 +1,62 @@
 import * as express from 'express';
-import * as cookieParser from 'cookie-parser';
 import * as httpProxy from 'http-proxy';
-import { URL } from 'url';
+import * as cookieParser from 'cookie-parser';
+import { noop } from 'lodash';
 
-import { health } from './health';
-import { environmentsByUrl } from './routingMap';
-import { createEnvNameGenerator } from './getEnvName';
-import { weightsByEnvironment } from './environments';
+import { health, setIsHealthy } from './health';
+import { redirectToHttps } from './redirectToHttps';
+import { getEnvForBranchPreview } from './branchPreviewer/getEnvForBranchPreview';
+import { getEnvForTrafficSplitting } from './trafficSplitter/getEnvForTrafficSplitting';
+import { EnvDetails } from './typings/environments.d';
 
-const getEnvName = createEnvNameGenerator(weightsByEnvironment);
+process.on('unhandledRejection', () => {
+  setIsHealthy(false);
+});
 
 const proxyServer = httpProxy.createProxyServer();
 
 const server = express();
 
 server.use('/health', health);
 
-// Redirect HTTP requests to HTTPS
-server.use((req, res, next) => {
-  const protocol = req.header('X-FORWARDED-PROTO');
-  const host = req.header('Host') || 'hollowverse.com';
-  if (protocol === 'http') {
-    const newURL = new URL(req.url, `https://${host}`);
-    res.redirect(newURL.toString());
-  } else {
-    next();
-  }
-});
+server.use(redirectToHttps);
 
 server.use(cookieParser());
 
+const trafficSplittingCookieName = 'env';
+const branchPreviewCookieName = 'branch';
+
 server.use(async (req, res) => {
-  const map = await environmentsByUrl;
-  let envName: string | undefined = req.cookies.env;
-  let envUrl: string | undefined;
-  if (!envName || map.get(envName) === undefined) {
-    envName = getEnvName.next().value;
+  let env: EnvDetails | void;
+
+  const branch = req.query.branch || req.cookies[branchPreviewCookieName];
+  if (branch) {
+    res.setHeader('X-Hollowverse-Requested-Environment', branch);
+
+    env = await getEnvForBranchPreview(branch).catch(noop);
+    if (env) {
+      res.cookie(branchPreviewCookieName, env.name, {
+        maxAge: 2 * 60 * 60 * 1000,
+      });
+    } else {
+      res.clearCookie(branchPreviewCookieName);
```

Comment:
When will this get executed? If there's no `env` for the requested branch, the cookie won't be set to begin with, so why do you clear it if it doesn't get set?

Reply:
What if it was set and the test environment was deleted afterwards? If we don't remove the cookie we will have to send a request to EB API on every request.